build_tools/cmake/iree_llvm.cmake

```diff
@@ -65,6 +65,8 @@ macro(iree_llvm_configure_bundled)
 
   set(LLVM_BINARY_DIR "${IREE_BINARY_DIR}/llvm-project")
   set(LLVM_TOOLS_BINARY_DIR "${LLVM_BINARY_DIR}/bin")
+  set(LLVM_RUNTIME_OUTPUT_INTDIR ${LLVM_BINARY_DIR}/bin)
```

Comment:
This change may not be correct. These tests are failing because the binary is in the wrong directory:
```
cmake --build ./build --target check-iree-dialects
```

Related change: https://reviews.llvm.org/D153373

Reply:
You need the one line change to lit.cfg.py here: https://github.com/openxla/iree/pull/14164/files#diff-98f8a8032bf8796b083cf1327985fc9841ceffd9707da93f579af3ab221994b9

Note also the changes to iree_llvm.cmake for the INTDIR.

(Sorry - thought I'd squeeze that in before the integrate picked it up from head(

Reply:
Actually, I need to revert the patch upstream that was causing the INTDIR changes: https://reviews.llvm.org/D153373

So you can cherrypick a revert if it helps this integrate.

Reply:
I took the change from llvm-project.